### ISSUE TITLE
Bug report: using tanh for logits is wrong

### ISSUE THREAD

**Issue by StKlaas** [macanv/BERT-BiLSTM-CRF-NER#159]
https://github.com/macanv/BERT-BiLSTM-CRF-NER/blob/c1aa61d9c095e621dd10c6756a1ae651eb1d54f6/bert_base/train/lstm_crf_layer.py#L147

Generally, the above line computes the logit for each tag but adding the 'tanh' function is wrong since it will map the logits to the interval [-1, 1], thus hurt the Viterbi decoding processing and the model performance. 
At first, I ran my model with the above line, the results indicated that softmax is better than crf. This is counterintuitive. After removing the 'tanh' function, the obtained results seemed more reasonable with crf surpass softmax.

**Comment by leonaxiongxin** [comment]
I ran into the same problem. When I used bert+crf with `tanh`, the loss function was much larger then bert. When `tanh` was removed, bert+crf performed better.

**Comment by HelloFranker** [comment]
great issue

**Comment by HelloFranker** [comment]
Great issue,。之前有issue提到了，有人实验了在他们的数据上，单存的Bert模型NER效果能达到95%，使用Bert_CRF效果变成了90%。这其中可能是tanh影响了Bert_CRF的效果，可以尝试将此处的tanh去掉，看一下Bert_CRF的模型效果。

**Comment by xmy7216** [comment]
> Great issue,。之前有issue提到了，有人实验了在他们的数据上，单存的Bert模型NER效果能达到95%，使用Bert_CRF效果变成了90%。这其中可能是tanh影响了Bert_CRF的效果，可以尝试将此处的tanh去掉，看一下Bert_CRF的模型效果。

您好，请问如果不使用tanh是直接去掉那一行吗？需要做其它改动吗？

**Comment by macanv** [comment]
with that issue https://github.com/macanv/BERT-BiLSTM-CRF-NER/issues/254, and new code version has remove tanh. THX eveybody!!!

**Comment by Heaven-zhw** [comment]
> I ran into the same problem. When I used bert+crf with `tanh`, the loss function was much larger then bert. When `tanh` was removed, bert+crf performed better.

Hello, after removing "tanh", which model preforms better in your experiments? In my progrom, bert+crf and bert+bilstm+crf both remove "tanh", and bert+crf performs better than using bilstm. Is that reasonable?

**Comment by leonaxiongxin** [comment]
> > I ran into the same problem. When I used bert+crf with `tanh`, the loss function was much larger then bert. When `tanh` was removed, bert+crf performed better.
> 
> Hello, after removing "tanh", which model preforms better in your experiments? In my progrom, bert+crf and bert+bilstm+crf both remove "tanh", and bert+crf performs better than using bilstm. Is that reasonable?
Yes, it happened to me at first. Maybe parameters in bilstm layer are not optimal. 
By the way, in my opinion, CRF and BiLSTM are not strictly comparable because because they are two different models. So I compared between  BiLSTM and BiLSTM-CRF to prove that CRF is effective for NER.

**Comment by leonaxiongxin** [comment]
> > I ran into the same problem. When I used bert+crf with `tanh`, the loss function was much larger then bert. When `tanh` was removed, bert+crf performed better.
> 
> Hello, after removing "tanh", which model preforms better in your experiments? In my progrom, bert+crf and bert+bilstm+crf both remove "tanh", and bert+crf performs better than using bilstm. Is that reasonable?

Yes, it happened to me at first. Maybe parameters in bilstm layer are not optimal.
By the way, in my opinion, CRF and BiLSTM are not strictly comparable because they are two different models. So I compared between BiLSTM and BiLSTM-CRF to prove that CRF is effective for NER.

**Comment by Heaven-zhw** [comment]
> > > I ran into the same problem. When I used bert+crf with `tanh`, the loss function was much larger then bert. When `tanh` was removed, bert+crf performed better.
> > 
> > 
> > Hello, after removing "tanh", which model preforms better in your experiments? In my progrom, bert+crf and bert+bilstm+crf both remove "tanh", and bert+crf performs better than using bilstm. Is that reasonable?
> 
> Yes, it happened to me at first. Maybe parameters in bilstm layer are not optimal.
> By the way, in my opinion, CRF and BiLSTM are not strictly comparable because they are two different models. So I compared between BiLSTM and BiLSTM-CRF to prove that CRF is effective for NER.

OK, thanks for your apply
